client/src/components/jobPosting/TimespanPicker.js

```diff
@@ -10,34 +10,30 @@ export class TimespanPicker extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      showFrom: moment(),
+      showFrom: null,
       showTo: null,
       error: ''
     }
     this.handleShowFromChange = this.handleShowFromChange.bind(this)
     this.handleShowToChange = this.handleShowToChange.bind(this)
   }
 
-  componentDidMount() {
-    this.props.addShowFrom(this.state.showFrom)
-  }
 
-  handleShowFromChange(date) {
+  async handleShowFromChange(date) {
     if (date !== undefined && (date === null || !this.state.showTo || date.isBefore(this.state.showTo))) {
       if (date === null) {
-        this.setState({
+        await this.setState({
           showFrom: null,
-          showTo: null,
-          error: ''
+          showTo: null
         })
-        this.props.addShowFrom({ date })
-        this.props.addShowTo({ date })
+        this.props.addShowFrom(null)
       } else {
-        this.setState({
+        await this.setState({
           showFrom: date,
           error: ''
         })
-        this.props.addShowFrom({ date })
+        const formattedDate = moment(date).format('YYYY-MM-DD')
```

Comment:
I think Date toLocaleString or similar was used else where. Might be best to be consistent.